Source/WebCore/page/ContextMenuContext.cpp

```diff
@@ -42,7 +44,18 @@ ContextMenuContext::ContextMenuContext(Type type, const HitTestResult& hitTestRe
     : m_type(type)
     , m_hitTestResult(hitTestResult)
     , m_event(event)
+    , m_hasEntireImage(false)
 {
+    auto imageURL = hitTestResult.absoluteImageURL();
+    if (imageURL.isEmpty() || hitTestResult.imageRect().isEmpty())
+        return;
+
+    auto* innerNodeFrame = hitTestResult.innerNodeFrame();
+    if (!innerNodeFrame)
+        return;
+
+    if (auto page = innerNodeFrame->page())
+        m_hasEntireImage = page->hasLocalDataForURL(imageURL);
```

Comment:
I think this would be better encapsulated as a `const` function on `WebCore::HitTestResult`. Then, we could have `m_hasEntireImage(hitTestResult.hasEntireImage())` in the constructor.

Not a big fan of the early turns in the constructor, since we would may have to refactor if we were to add more properties.

Reply:
Excellent point, will fix.